shared/elements/gaia_grid/js/items/grid_item.js

```diff
@@ -514,7 +514,14 @@
 
         var nameEl = document.createElement('span');
         nameEl.className = 'title';
-        nameEl.textContent = this.name;
+
+        if (this.name && this.name.then) {
```

Comment:
If you want to have some kind of ```renderName``` or similar method that populates the DOM I would be fine with that as well. Having the method override the base class might be more clean than having the current if/else hook.